### PR TITLE
Updates to Dockerfile

### PR DESCRIPTION
**Pull request by DenisCarriere**
- Add environment variables from PR https://github.com/Project-OSRM/osrm-frontend/pull/244
- Use [`Alpine Node`](https://github.com/mhart/alpine-node) docker image
- Use `yarn` to install
- Add comments to Docker instructions

### CODE REVIEW

**Comment by DenisCarriere** [comment]
**Extra**

You could drop `docker/hooks` and place the `Dockerfile` & `.dockerignore` files in the root directory. There's no "gymnastics" needed (maybe with the new `replace` PR this will work).

CC: @daniel-j-h @danpat

**Comment by freenerd** [comment]
Thanks for the PR. Env vars and comments look great.

Sorry, but I think we will not switch the project to yarn. I understand it's merrits, but right now everything in the project-osrm universe is using npm and staying consistent is more important.

Likely we'd like to stay on the official alpine image, for reasons of security but also reduced complexity. By moving `COPY . /src` behind the node install, the main pain point of slow local development via docker should be mitigated already though, not?

**Comment by DenisCarriere** [comment]
@freenerd 👍 No worries, the main reason for this PR was to define the new Environment variables, I've dropped the `Yarn` build and removed the `yarn.lock` from this PR, this should be a very low-impact PR now.

**Comment by DenisCarriere** [comment]
The reason behind:
```dockerfile
# Install app dependencies
COPY package.json /src
RUN npm install

# Bundle app source
COPY . /src
```
Is whenever you change any file in the repo, the Docker image needs to be entirely re-built including rebuilding all the dependencies. If you don't change any dependencies in the `package.json` then the `RUN npm install` doesn't have to be rebuilt, this step usually takes a while and if you can skip this step it's faster to create a Docker container for dev purposes.

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 
+# Create app directory
 RUN mkdir /src
 COPY . /src
 WORKDIR /src
 
+# Install app dependencies
 RUN apk add --no-cache sed nodejs && \
     cd /src && \
     npm install
```
Comment (DenisCarriere) [inline docker/Dockerfile]: @freenerd This `COPY . /src` should declared after `npm install`.

Not an urgent, it's just faster to build the Docker container if you don't have to do `npm install` for every single minor change.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+# Enables customized options using environment variables
 ENV BACKEND='http://localhost:5000'
+ENV CENTER='38.8995,-77.0269'
+ENV ZOOM='13'
+ENV LANGUAGE='en'
+ENV LABEL='Car (fastest)'
 
+# Create app directory
 RUN mkdir /src
 COPY . /src
```
Comment (DenisCarriere) [inline docker/Dockerfile]: @freenerd This `COPY . /src` should declared after `npm install`.

Not an urgent, it's just faster to build the Docker container if you don't have to do npm install for every single minor change.

Comment (freenerd) [reply docker/Dockerfile]: Yeah, I'm good on doing that 😄 That's what I tried to say in https://github.com/Project-OSRM/osrm-frontend/pull/245#issuecomment-313698478 😉

Comment (DenisCarriere) [reply docker/Dockerfile]: 👍 Sounds like a plan, feel free to change/modify this PR before merging, or close it.

**Comment by DenisCarriere** [comment]
@freenerd 👍

**Pull request merged**